File: samples/lambda-function-urls/python/src/handler.py
"""
Lambda Function URL Handler

Demonstrates Lambda Function URLs - HTTPS endpoints directly on Lambda functions.
Returns information about the incoming request.
"""

import json


def handler(event, context):
    """
    Handle HTTP requests via Lambda Function URL.

    Args:
        event: Request event from Function URL
        context: Lambda context

    Returns:
        HTTP response with request details
    """
    # Extract request details
    http_method = event.get("requestContext", {}).get("http", {}).get("method", "UNKNOWN")
    path = event.get("requestContext", {}).get("http", {}).get("path", "/")
    query_params = event.get("queryStringParameters") or {}
    headers = event.get("headers") or {}

    # Parse body if present
    body = event.get("body")
    if body:
        try:
            body = json.loads(body)
        except (json.JSONDecodeError, TypeError):
            pass  # Keep as string if not JSON

    # Build response
    response_body = {
        "message": "Hello from Lambda Function URL!",
        "request": {
            "method": http_method,
            "path": path,
            "queryParams": query_params,
            "body": body
        },
        "functionName": context.function_name if context else "unknown"
    }

    return {
        "statusCode": 200,
        "headers": {
            "Content-Type": "application/json"
        },
        "body": json.dumps(response_body)
    }

File: samples/elb-load-balancing/javascript/handler.js
/**
 * ELB Lambda handlers.
 *
 * These handlers respond to ALB requests with simple messages.
 * Response format follows ALB Lambda integration requirements.
 */

// Handler for /hello1 path
module.exports.hello1 = async (event) => {
    console.log('hello1 invoked:', JSON.stringify(event));
    return {
        isBase64Encoded: false,
        statusCode: 200,
        statusDescription: '200 OK',
        headers: {
            'Content-Type': 'application/json'
        },
        body: JSON.stringify({ message: 'Hello 1', path: '/hello1' })
    };
};

// Handler for /hello2 path
module.exports.hello2 = async (event) => {
    console.log('hello2 invoked:', JSON.stringify(event));
    return {
        isBase64Encoded: false,
        statusCode: 200,
        statusDescription: '200 OK',
        headers: {
            'Content-Type': 'application/json'
        },
        body: JSON.stringify({ message: 'Hello 2', path: '/hello2' })
    };
};

// Default handler
module.exports.handler = async (event) => {
    console.log('default handler invoked:', JSON.stringify(event));
    return {
        isBase64Encoded: false,
        statusCode: 200,
        statusDescription: '200 OK',
        headers: {
            'Content-Type': 'application/json'
        },
        body: JSON.stringify({ message: 'Hello from ELB Lambda' })
    };
};
